File: annotationprocessingtesting/src/main/java/com/tickaroo/tikxml/annotationprocessing/textcontent/autovalue/Day.java
package com.tickaroo.tikxml.annotationprocessing.textcontent.autovalue;

import com.google.auto.value.AutoValue;
import com.tickaroo.tikxml.annotation.TextContent;
import com.tickaroo.tikxml.annotation.Xml;

/**
 * @author Hannes Dorfmann
 */
@Xml
@AutoValue
public abstract class Day {

  @TextContent public abstract String name();
}

File: annotationprocessortesting-kt/src/main/java/com/tickaroo/tikxml/annotationprocessing/textcontent/autovalue/Day.kt
package com.tickaroo.tikxml.annotationprocessing.textcontent.autovalue

import com.google.auto.value.AutoValue
import com.tickaroo.tikxml.annotation.GenericAdapter
import com.tickaroo.tikxml.annotation.TextContent
import com.tickaroo.tikxml.annotation.Xml

/**
 * @author Hannes Dorfmann
 */
@Xml
@AutoValue
@GenericAdapter
abstract class Day {
    @TextContent
    abstract fun name(): String
}
